FrameworkSystem/Utilities/halo.py

```diff
@@ -0,0 +1,687 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=unsubscriptable-object
+""" Changed for DIRAC. Beautiful terminal spinners in Python. Source: https://github.com/manrajgrover/halo and dependens
+"""
```

Comment:
... can we? There's no license?

Reply:
Not without copying the copyright notice and MIT license text over.

Regardless, why not use `import halo`?

Reply:
If I add the license text to the `halo.py` file, will that be enough?
I changed the code a bit for ease of use and threw away things that won't be used.
The `colorama` and `termcolor` packages imported by the original `halo`, I almost did not change.
In fact, I was also guided by what to be less dependent on third-party packages.

Reply:
> If I add the license text to the halo.py file, will that be enough?

Legally, yes. Though you would also need the `colorama` and `termcolor` licenses as well as you've copied those too.

>  In fact, I was also guided by what to be less dependent on third-party packages.

Why? DIRAC has hundreds of dependencies already and these three are widely used and pure python so they're easy to deploy. If we don't want to add the dependency then I would argue we should use static black-and-white output instead.

Reply:
can we also settle on one ? I think I have seen `tqdm` used somewhere DIRAC related

Reply:
`tqdm` provide progress bar and `halo` provide spinners. I think we need both for DIRAC.